src/main/java/org/jenkinsci/test/acceptance/controller/WinstoneController.java

```diff
@@ -43,27 +49,55 @@ public void addJavaOpt(String javaOpt) {
         }
     }
 
-    protected final int httpPort;
+    protected int httpPort;
 
     @Inject
     public WinstoneController(Injector i) {
-        this(i, IOUtil.randomTcpPort());
+        this(i, IOUtil.randomTcpPort()); // TODO Change to 0 when Jenkins has https://github.com/jenkinsci/jenkins/pull/5928
     }
 
     public WinstoneController(Injector i, int httpPort) {
         super(i);
         this.httpPort = httpPort;
     }
 
+    @Override
+    public void startNow() throws IOException {
+        super.startNow();
+        if (this.httpPort == 0) {
+            this.httpPort = readPort(portFile);
+        }
+    }
+
+    @Override
+    protected void onReady() throws IOException {
+        if (this.httpPort == 0) {
+            this.httpPort = readPort(portFile);
+        }
+    }
+
+    private static int readPort(File portFile) throws IOException {
+        String s = FileUtils.readFileToString(portFile, StandardCharsets.UTF_8);
```

Comment:
incorrect charset is being used here?
https://github.com/jenkinsci/winstone/pull/174#discussion_r752472871

Reply:
`[0-9]+` so who cares?

Reply:
some people extend the ATH to test their workflows and and some people use that funky IBM thingy....
I have no idea if those users overlap or not - but this is a trivial thing to fix?

Reply:
z/OS only for agents I think?